dashboard/app/models/course_version.rb

```diff
@@ -151,6 +152,14 @@ def recommended?(locale_code = 'en-us')
     latest_stable_version == content_root
   end
 
+  def self.course_versions_with_student_progress(student_unit_ids, user)
```

Comment:
This API was a little confusing to me.  I think it's because, from the name, I expected it to look at progress and the parameter name `student_unit_ids` doesn't indicate that it has to do with progress.

I'm wondering if maybe this API can be something more generic like `course_versions_containing_units` and just take an array of `unit_ids`?